tools/marketplace/helper/bundleHelper.go

```diff
@@ -0,0 +1,311 @@
+		klog.Errorf("Failed to unmarshal CSV data", err)
+		return err
+	}
+
+	// use k8s json unmarshaller for CRDs for filling metadata correctly
+	crds, err := yaml2.YAMLToJSON([]byte(bundle.Data.CRDs))
+	if err != nil {
+		klog.Errorf("Failed to convert CRD data to json format", err)
+		return err
+	}
+	if err := json.Unmarshal([]byte(crds), &bh.CRDs); err != nil {
+		klog.Errorf("Failed to unmarshal CRD data", err)
+		return err
+	}
+
+	return nil
+}
+
+func (bh *BundleHelper) getRepoBundles(bundles []*datastore.RegistryMetadata) ([]*datastore.RegistryMetadata, error) {
+	list := make([]*datastore.RegistryMetadata, len(bundles))
+	i := 0
+	for j := 0; j < len(bundles); j++ {
+		bundle := bundles[j]
+
+		if bh.repo == bundle.Repository {
+			metadata := &datastore.RegistryMetadata{
+				Namespace:  bundle.Namespace,
+				Repository: bundle.Repository,
+
+				// 'Default' points to the latest release pushed.
+				Release: bundle.Release,
+
+				// Getting 'Digest' would require an additional call to the app
+				// registry, so it is being defaulted.
+			}
+
+			list[i] = metadata
+			i++
+		} //copy bundle info onlky if repo matches
+	}
+
+	return list, nil
+}
+
+//AddOldManifests - downloads old CRDs and CSVs to outDir with respect to currentCSVVersion
+func (bh *BundleHelper) AddOldManifests(outDir string, currentCSVVersion string) error {
+
+	if err := bh.addOldCRDs(outDir); err != nil {
+		klog.Errorf("Failed to add old CRDs to %s", outDir, err)
+		return err
+	}
+
+	if err := bh.addOldCSVs(outDir, currentCSVVersion); err != nil {
+		klog.Errorf("Failed to add old CSVs to %s", outDir, err)
+		return err
+	}
+
+	return nil
+
+}
+
+func (bh *BundleHelper) addOldCRDs(outDir string) error {
+
+	currentVersion := v1.CDIGroupVersionKind.Version
+	for _, crd := range bh.CRDs {
+		if crd.Spec.Version == currentVersion {
+			// the current version wil be generated
+			continue
+		}
+		// write old CRD to the out dir
+		bytes, err := json.Marshal(crd)
+		if err != nil {
+			klog.Errorf("Failed to marshal old CRDs ", err)
+			return err
+		}
+		filename := fmt.Sprintf("%v/%v-%v.crd.yaml", outDir, crd.Name, crd.Spec.Version)
+		err = ioutil.WriteFile(filename, bytes, 0644)
+		if err != nil {
+			klog.Errorf("Failed to write old CRDs in %s", filename, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (bh *BundleHelper) addOldCSVs(outDir string, currentCSVVersion string) error {
+
+	for _, csv := range bh.CSVs {
+
+		version := GetCSVVersion(csv)
+
+		if version == currentCSVVersion {
+			// the current version wil be generated
+			continue
+		}
+		// write old CSV to the out dir
+		bytes, err := yaml.Marshal(csv)
+		if err != nil {
+			klog.Errorf("Failed to marshal old CSVs in ", err)
+			return err
+		}
+		filename := fmt.Sprintf("%v/%v.csv.yaml", outDir, GetCSVName(csv))
+		err = ioutil.WriteFile(filename, bytes, 0644)
+		if err != nil {
+			klog.Errorf("Failed to write old CSVs in %s", filename, err)
```

Comment:
Missing %v for error

Reply:
Arh, apparently I consistently ignored all the errors ;( that copy paste...